client/src/main/java/org/apache/celeborn/client/ShuffleClient.java

```diff
@@ -252,6 +257,8 @@ public abstract CelebornInputStream readPartition(
       ExceptionMaker exceptionMaker,
       ArrayList<PartitionLocation> locations,
       ArrayList<PbStreamHandler> streamHandlers,
+      Map<String, Set<PushFailedBatch>> failedBatchSetMap,
```

Comment:
I think this was discussed and I am missing some reference.
Given the design is abandon the current stage (and the parent stage(s) shuffle output when relevant) - why are we tracking `PushFailedBatch` ?
It will entirely get abandoned anyway, right ?

Reply:
PushFailedBatch is for those batches need retry when push encounter exception, and same batch may occur in different chunks, tracking PushFailedBatches can avoid duplicate read.

Reply:
I am trying to understand why this is specific to this PR.
For a non skewed stage, we handle this right ? Won't the same not apply here ?

Reply:
> For a non skewed stage, we handle this right
In a non-skewed stage, there is no need for this, as the reducer can read data by map range, allowing for the deduplication of identical batches when the reducer processes the entire dataset from the map task. However, in a skewed stage, the reducer reads only partial data in chunks, which may originate from all map tasks. In this scenario, identical batches may appear in different chunks, making it difficult for the reducer to deduplicate them unless it is aware of which batches shouldn't be read, that's why all map tasks should tell LifecycleManager failedBatches which can't be read.